src/net/server/bytearrstream.rs

```diff
@@ -0,0 +1,456 @@
+use crate::{error::BoxError, ff::FieldTypeStr, net::MpcHelperServerError};
+use async_trait::async_trait;
+use axum::extract::{BodyStream, FromRequest, Query, RequestParts};
+use futures::{ready, Stream};
+use hyper::body::{Bytes, HttpBody};
+use pin_project::pin_project;
+use std::collections::VecDeque;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// the size of the field type specified in the query param of the request
+/// Possible sizes are are currently from:
+/// * `fp2`
+/// * `fp31`
+/// * `fp32_bit_prime`
+struct FieldSize {
```

Comment:
this transformation is lossy because it forgets about the actual field requested by the user. if we have two fields that happen to be the same size, we won't be able to resolve this ambiguity. Besides, we need actual field type to construct the protocol

Reply:
there is actually more evidence that using field size as alignment for input streams is not correct. For protocols that take replicated shares as input, the correct alignment is 2 * field_size